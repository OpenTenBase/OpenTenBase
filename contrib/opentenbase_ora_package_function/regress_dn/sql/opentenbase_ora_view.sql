\c opentenbase_ora_package_function_regression_ora
\set ECHO none
set client_min_messages TO error;
CREATE EXTENSION IF NOT EXISTS opentenbase_ora_package_function;
set client_min_messages TO default;

-- =========================================================================
-- DBA_ALL_TABLES
-- ALL_ALL_TABLES
-- USER_ALL_TABLES
-- =========================================================================

SELECT * FROM DBA_TABLES limit 0;
SELECT * FROM ALL_TABLES limit 0;
SELECT * FROM USER_TABLES limit 0;

create table v_all_tables_test_1(id int);
create table v_all_tables_test_2(id int);
insert into v_all_tables_test_2 select n from generate_series(1,100) n;
analyze v_all_tables_test_2;

create temporary table v_all_tables_test_3(id int);
insert into v_all_tables_test_3 select n from generate_series(1,100) n;
analyze v_all_tables_test_3;

SELECT table_name, duration is null, LAST_ANALYZED is null FROM ALL_TABLES WHERE TABLE_NAME ilike 'v_all_tables_test%' order by 1;
SELECT table_name, duration is null, LAST_ANALYZED is null FROM DBA_TABLES WHERE TABLE_NAME ilike 'v_all_tables_test%' order by 1;
SELECT table_name, duration is null, LAST_ANALYZED is null FROM USER_TABLES WHERE TABLE_NAME ilike 'v_all_tables_test%' order by 1;

drop table v_all_tables_test_3;
drop table v_all_tables_test_2;
drop table v_all_tables_test_1;
