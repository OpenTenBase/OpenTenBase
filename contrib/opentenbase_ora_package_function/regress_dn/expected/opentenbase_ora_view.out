\c opentenbase_ora_package_function_regression_ora
\set ECHO none
 OWNER | SCHEMA_NAME | TABLE_NAME | TABLESPACE_NAME | CLUSTER_NAME | IOT_NAME | STATUS | PCT_FREE | PCT_USED | INI_TRANS | MAX_TRANS | INITIAL_EXTENT | NEXT_EXTENT | MIN_EXTENTS | MAX_EXTENTS | PCT_INCREASE | FREELISTS | FREELIST_GROUPS | LOGGING | BACKED_UP | NUM_ROWS | BLOCKS | EMPTY_BLOCKS | AVG_SPACE | CHAIN_CNT | AVG_ROW_LEN | AVG_SPACE_FREELIST_BLOCKS | NUM_FREELIST_BLOCKS | DEGREE | INSTANCES | CACHE | TABLE_LOCK | SAMPLE_SIZE | LAST_ANALYZED | PARTITIONED | IOT_TYPE | TEMPORARY | SECONDARY | NESTED | BUFFER_POOL | FLASH_CACHE | CELL_FLASH_CACHE | ROW_MOVEMENT | GLOBAL_STATS | USER_STATS | DURATION | SKIP_CORRUPT | MONITORING | CLUSTER_OWNER | DEPENDENCIES | COMPRESSION | COMPRESS_FOR | DROPPED | READ_ONLY | SEGMENT_CREATED | RESULT_CACHE 
-------+-------------+------------+-----------------+--------------+----------+--------+----------+----------+-----------+-----------+----------------+-------------+-------------+-------------+--------------+-----------+-----------------+---------+-----------+----------+--------+--------------+-----------+-----------+-------------+---------------------------+---------------------+--------+-----------+-------+------------+-------------+---------------+-------------+----------+-----------+-----------+--------+-------------+-------------+------------------+--------------+--------------+------------+----------+--------------+------------+---------------+--------------+-------------+--------------+---------+-----------+-----------------+--------------
(0 rows)

 OWNER | SCHEMA_NAME | TABLE_NAME | TABLESPACE_NAME | CLUSTER_NAME | IOT_NAME | STATUS | PCT_FREE | PCT_USED | INI_TRANS | MAX_TRANS | INITIAL_EXTENT | NEXT_EXTENT | MIN_EXTENTS | MAX_EXTENTS | PCT_INCREASE | FREELISTS | FREELIST_GROUPS | LOGGING | BACKED_UP | NUM_ROWS | BLOCKS | EMPTY_BLOCKS | AVG_SPACE | CHAIN_CNT | AVG_ROW_LEN | AVG_SPACE_FREELIST_BLOCKS | NUM_FREELIST_BLOCKS | DEGREE | INSTANCES | CACHE | TABLE_LOCK | SAMPLE_SIZE | LAST_ANALYZED | PARTITIONED | IOT_TYPE | TEMPORARY | SECONDARY | NESTED | BUFFER_POOL | FLASH_CACHE | CELL_FLASH_CACHE | ROW_MOVEMENT | GLOBAL_STATS | USER_STATS | DURATION | SKIP_CORRUPT | MONITORING | CLUSTER_OWNER | DEPENDENCIES | COMPRESSION | COMPRESS_FOR | DROPPED | READ_ONLY | SEGMENT_CREATED | RESULT_CACHE 
-------+-------------+------------+-----------------+--------------+----------+--------+----------+----------+-----------+-----------+----------------+-------------+-------------+-------------+--------------+-----------+-----------------+---------+-----------+----------+--------+--------------+-----------+-----------+-------------+---------------------------+---------------------+--------+-----------+-------+------------+-------------+---------------+-------------+----------+-----------+-----------+--------+-------------+-------------+------------------+--------------+--------------+------------+----------+--------------+------------+---------------+--------------+-------------+--------------+---------+-----------+-----------------+--------------
(0 rows)

 SCHEMA_NAME | TABLE_NAME | TABLESPACE_NAME | CLUSTER_NAME | IOT_NAME | STATUS | PCT_FREE | PCT_USED | INI_TRANS | MAX_TRANS | INITIAL_EXTENT | NEXT_EXTENT | MIN_EXTENTS | MAX_EXTENTS | PCT_INCREASE | FREELISTS | FREELIST_GROUPS | LOGGING | BACKED_UP | NUM_ROWS | BLOCKS | EMPTY_BLOCKS | AVG_SPACE | CHAIN_CNT | AVG_ROW_LEN | AVG_SPACE_FREELIST_BLOCKS | NUM_FREELIST_BLOCKS | DEGREE | INSTANCES | CACHE | TABLE_LOCK | SAMPLE_SIZE | LAST_ANALYZED | PARTITIONED | IOT_TYPE | TEMPORARY | SECONDARY | NESTED | BUFFER_POOL | FLASH_CACHE | CELL_FLASH_CACHE | ROW_MOVEMENT | GLOBAL_STATS | USER_STATS | DURATION | SKIP_CORRUPT | MONITORING | CLUSTER_OWNER | DEPENDENCIES | COMPRESSION | COMPRESS_FOR | DROPPED | READ_ONLY | SEGMENT_CREATED | RESULT_CACHE 
-------------+------------+-----------------+--------------+----------+--------+----------+----------+-----------+-----------+----------------+-------------+-------------+-------------+--------------+-----------+-----------------+---------+-----------+----------+--------+--------------+-----------+-----------+-------------+---------------------------+---------------------+--------+-----------+-------+------------+-------------+---------------+-------------+----------+-----------+-----------+--------+-------------+-------------+------------------+--------------+--------------+------------+----------+--------------+------------+---------------+--------------+-------------+--------------+---------+-----------+-----------------+--------------
(0 rows)

     TABLE_NAME      | ?column? | ?column? 
---------------------+----------+----------
 V_ALL_TABLES_TEST_1 | t        | t
 V_ALL_TABLES_TEST_2 | f        | f
 V_ALL_TABLES_TEST_3 | f        | f
(3 rows)

     TABLE_NAME      | ?column? | ?column? 
---------------------+----------+----------
 V_ALL_TABLES_TEST_1 | t        | t
 V_ALL_TABLES_TEST_2 | f        | f
 V_ALL_TABLES_TEST_3 | f        | f
(3 rows)

     TABLE_NAME      | ?column? | ?column? 
---------------------+----------+----------
 V_ALL_TABLES_TEST_1 | t        | t
 V_ALL_TABLES_TEST_2 | f        | f
 V_ALL_TABLES_TEST_3 | f        | f
(3 rows)
